Documentation/ApiOverview/Database/Connection/Index.rst

```diff
@@ -6,356 +6,479 @@
 Connection
 ==========
 
-An instance of class :php:`TYPO3\CMS\Core\Database\Connection` is retrieved from the
-:ref:`ConnectionPool <database-connection-pool>` by calling `->getConnectionForTable()`
-and handing over the table name a query should executed on.
+.. contents:: **Table of Contents**
+   :local:
 
-The class extends the basic Doctrine DBAL `Doctrine\DBAL\Connection` class and is mainly
-used internally within the TYPO3 framework to establish, maintain and terminate
-connections to single database endpoints. Those internal methods are not scope of this
-documentation since an extension developer usually doesn't have to deal with that.
 
-For an extension developer however, the class provides a list of "short-hand" methods
-that allow dealing with "simple" query cases, without the complexity of the
-:ref:`QueryBuilder <database-query-builder>`. Using those methods typically ends up in
-rather short and easily readable code. The methods have in common that they support only
-"equal" comparisons in `WHERE` conditions, that all fields and values are fully quoted
-automatically and the created queries are executed right away.
+.. _database-connection-introduction:
 
-.. note::
+Introduction
+============
 
-    The `Connection` object is designed to work on a single table only. If queries to multiple
-    tables should be performed, the object must not be re-used. Instead, a single `Connection`
-    instance should be retrieved via `ConnectionPool` per target table. However, it is allowed
-    to use one `Connection` object for multiple queries to the same table.
+The :php:`TYPO3\CMS\Core\Database\Connection` class extends the basic Doctrine
+DBAL :php:`Doctrine\DBAL\Connection` class and is mainly used internally in
+TYPO3 to establish, maintain and terminate connections to single database
+endpoints. These internal methods are not the scope of this documentation, since
+an extension developer usually does not have to deal with them.
 
+However, for an extension developer, the class provides a list of short-hand
+methods that allow you to deal with query cases without the complexity
+of the :ref:`query builder <database-query-builder>`. Using these methods
+usually ends up in rather short and easy-to-read code. The methods have in common
+that they only support "equal" comparisons in :sql:`WHERE` conditions, that all
+fields and values are automatically fully quoted, and that the created queries
+are executed right away.
 
-insert()
-========
+..  note::
+    The :php:`Connection` object is designed to work on a single table only. If
+    queries are performed on multiple tables, the object must not be reused.
+    Instead, a single :php:`Connection` instance per target table should be
+    retrieved via :ref:`ConnectionPool <database-connection-pool>`. However, it
+    is allowed to use one :php:`Connection` object for multiple queries on the
```

Comment:
Are you sure about this? I think I rember there was something a while back that you should never reuse it...

Reply:
Yup, I am sure, the query builder has state and should not be used multiple times, the connection is only bound to the table and can be used multiple times.